[optim] bugfix when all parameters have no grad

wanchaol opened the pull request:
Summary: This fix the bug introduced during refactoring optimizers https://github.com/pytorch/pytorch/pull/50411. When all parameters have no grads, we should still allows `beta` like hyper params to be defined.

Differential Revision: D26699827

albanD commented on test/test_optim.py:
```diff
@@ -633,6 +633,15 @@ def test_duplicate_params_in_param_group(self):
+    def test_no_grad_for_all_params(self):
+        param = torch.randn(5, 5, requires_grad=False)
+
+        for optim_ctr in [optim.AdamW, optim.Adadelta]:
```
nit: Could you actually extend this to all of our optimizers?

albanD reviewed (approved):
Thanks for the fix!

Commit:
[optim] bugfix when all parameters have no grad

Summary: This fix the bug introduced during refactoring optimizers https://github.com/pytorch/pytorch/pull/50411. When all parameters have no grads, we should still allows `beta` like hyper params to be defined.

Reviewed By: ngimel

Differential Revision: D26699827

fbshipit-source-id: fbe58186ae17e131abf6cdaf6eabbe32e351fe94
